File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        lp = 0
        rp = len(nums)-1
        while(lp<rp):
            sum = nums[lp]+nums[rp]
            if(sum<target):
                lp+=1
            elif sum>target:
                rp-=1
            if(sum==target):
                return [lp+1,rp+1]
        return null
        

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int lp = 0;
        int rp = nums.length-1;
        while(lp<rp)
        {
            int res=nums[rp]+nums[lp];
            if(res<target)
            {
                lp+=1;
            }
            else if(res>target)
            {
                rp-=1;
            }
            if(res==target)
            {
                return new int[]{lp+1,rp+1};
            }
        }
        return new int[]{};
    }
}